### ISSUE TITLE
ImageList#append hangs

### ISSUE THREAD

**Issue by shviller** [gemhome/rmagick#202]
As instructed, I closed the issue at https://github.com/rmagick/rmagick/issues/129. I take it I should reopen it here, so here it is:

`ImageList#append` and other methods that output a single `Image` (such as `#average`) hang when used after adding more than one image via `+=`, `#concat` or `<<`. For example:

```
require 'rmagick'

x = Magick::Image.new(1,1)
il = Magick::ImageList.new
il << x << x
il.append(false)
```

Reproduced on two Windows 7 x64 boxes, a NixOS one and an Android phone running Lil Debi in chroot. rmagick 2.13.4 on all of them. ImageMagick versions: 6.8.9-6 Q16 x86 on Windows, 6.8.9-9 Q16 arm on the phone, 6.9.0-0 Q16 x86-64 on NixOS. Ruby 1.9.3 everywhere, 64-bit everywhere except the phone.

**Comment by vassilevsky** [comment]
Could you test at least one of the installations with the up-to-date RMagick 2.15? Changes were made to the C extension code that might help.

**Comment by shviller** [comment]
Confirmed with RMagick 2.15 on one Windows box and the NixOS one.

**Comment by u338steven** [comment]
@shviller Thanks for your report.
I can reproduce the problem. I will look into this.

Workaround:
As your example, if *identical instances* are concatenated, their methods(e.g. `ImageList#append`) hang.
Therefore, the following does not hang:

```rb
require 'rmagick'

x = Magick::Image.new(1,1)

il = Magick::ImageList.new
il << x << x.clone
il.append(false)
```

**Comment by vassilevsky** [comment]
I've released version 2.15.1 with the fix.
